### PR TITLE
Add pager option for customizing pager without environment variables

### PR DESCRIPTION
**Pull request by pbzweihander**
It would be great if we can configure `delta` with git config (or CLI args) only, without using environment variables.

### CODE REVIEW

**Comment by dandavison** [comment]
Hi @pbzweihander, thanks very much for this! I agree with you that this should exist; I believe I've wanted to add it several times but never quite got around to it.

File: src/main.rs [hunk 1]
```diff
@@ -82,7 +82,8 @@ fn run_app() -> std::io::Result<i32> {
         return Ok(0);
     }
 
-    let mut output_type = OutputType::from_mode(config.paging_mode, None, &config).unwrap();
+    let mut output_type =
+        OutputType::from_mode(config.paging_mode, config.pager.clone(), &config).unwrap();
```
Comment (dandavison) [inline src/main.rs:86]: Rust question (I've already merged the PR so this is not blocking, I'm just curious): so here we're `clone`ing an `Option<String>`. Do you have a view on whether that is preferable to passing `Option<&str>` and creating a `String` later as the code did previously? For example I think https://github.com/dandavison/delta/commit/bf277caa65bfee3dc51fdcfd3866c5c6afb3a560 is a commit which does that.

Comment (pbzweihander) [reply src/main.rs:86]: I thought that `to_string` (in https://github.com/dandavison/delta/blob/pbzweihander-configable-pager-str/src/bat_utils/output.rs#L74) will always clone the string. So I thought if `from_mode` receives an (already) owned string, the caller can decide whether clone or give ownership (and not clone).

Comment (dandavison) [reply src/main.rs:86]: I see, thanks! Yes, maybe, if we are really trying to optimize the code, there's a way to use `&str` internally in `from_mode` without converting to an owned `String`. But that's of course all stuff that existed before your changes. Thanks again!

**Pull request merged**